### PR TITLE
Finish prediction logics.

### LINKED ISSUE #3: Task 2: Backend service
We need to finish following tasks in backend service.

1. Web API for front-end. #9 
2. Predict with trained two-weeks-model.
3. Use database to cache stock list and predictions.

### LINKED ISSUE #6: Task 1.2: Let the model predict the price after 2 weeks
Established a code example to support the prediction we want.

Please see this branch for details: https://github.com/jingedawang/StockPredictor/tree/setup/algorithm
There are still some TODOs need to be solved in the code.

### LINKED ISSUE #9: Task 2.1: Web API for front-end
Web API provides following interfaces with HTTP protocol.

Interface 1: Get stock list.
```
Url: /stock/list
Parameter: None
Response: A JSON string. Example: 
[
	{
		"id": "600479",
		"pinyin": "QJYY",
		"name": "千金药业"
	},
	{
		"id": "600480",
		"pinyin": "LYGF",
		"name": "凌云股份"
	},
	{
		"id": "600481",
		"pinyin": "SLJN",
		"name": "双良节能"
	}
]
```

Interface 2: Predict.
```
Url: /stock/<id>
Parameter: <id>: The id of the stock.
Response: A JSON string containing both history prices and predicted price. Example:
{
	"id": "600000",
	"pinyin": "PFYH",
	"name": "浦发银行",
	"qlib_id": "SH600000",
	"enname": null,
	"history": [
		{
			"2020-08-03": 10.40999984741211
		},
		{
			"2020-08-04": 10.65999984741211
		},
		{
			"2020-08-05": 10.510001182556153
		},
		{
			"2020-08-06": 10.529999732971192
		},
		{
			"2020-08-07": 10.430000305175782
		}
	],
	"predict": 10.01
}
```

### LINKED ISSUE #30: Monitor the requests
Should monitor and save the request information to know how the service is used by the users.

Not sure if there are any good tools could be used.

### PR DESCRIPTION
**Pull request by jingedawang**
+ Do a full prediction when setting up the service.
+ Do a incremental prediction for today after the data update work in the evening.
+ Save trained model to 'stock_predictor' experiement and 'two_weeks_model' recorder.
+ Load the trained model in the predict script.
+ Cache the predictions in the database. Always using the cached data for frontend request.
+ Use {date: price} for prediction response.

### CODE REVIEW

**Review by sanzgiri — DISMISSED** [review]
Looks good. I am working on changes to support predictions for US stocks

**Comment by jingedawang** [comment]
> Looks good. I am working on changes to support predictions for US stocks

Great! We are approaching our goal.

File: stock_predictor/app.py [hunk 1]
```diff
@@ -67,19 +69,21 @@ def predict(id: str):
         "enname": "Shanghai Pudong Development Bank Co.,Ltd.",
         "history": [
```
Comment (y-vonn) [inline stock_predictor/app.py:70]: I don't quite understand... Is this writing test data in?

**Review by y-vonn — APPROVED** [review]

File: stock_predictor/app.py (hunk 1 above)
Comment (jingedawang) [reply stock_predictor/app.py:70]: This is comment. Document comment in Python is wrapped like
```python
"""
This is comment
"""
```

**Pull request merged**